src/compile/data/bin.ts

```diff
@@ -144,6 +144,9 @@ export class BinNode extends DataFlowNode {
     return out;
   }
 
+  public hash() {
+    return hash(this.bins);
```

Comment:
Could there be problems since the node type is not part of the hash?

Reply:
Good point, I'll include it in the hash. 
What's the best way though? Adding a `nodeType` key?

Maybe something like
`return hash({nodeType: "BinNode", ...this.bins});`

Reply:
Or just type + hash